src/node/desktop/src/main/session-launcher.ts

```diff
@@ -191,4 +193,34 @@ export class SessionLauncher {
   closeAllSatellites(): void {
     console.log('CloseAllSatellites not implemented');
   }
+
+  getLauncherToken(): string {
+    if (!this.launcherToken) {
+      this.launcherToken = generateShortenedUuid();
+    }
+    return this.launcherToken;
+  }
+
+  getPort(): string {
+    if (!this.port) {
+      // Use a random-ish port number to avoid collisions between different
+      // instances of rdesktop-launched rsessions; not a cryptographically
+      // secure technique so don't copy/paste for such purposes.
+      const base = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
+      this.port = ((base % 40000) + 8080).toString();
+
+      // recalculate the local peer and set RS_LOCAL_PEER so that
+      // rsession and it's children can use it
+      if (process.platform === 'win32') {
+        const localPeer = '\\\\.\\pipe\\' + this.port + '-rsession';
+        setenv('RS_LOCAL_PEER', localPeer);
+      }
+    }
+    return this.port;
+  }
+
+  newPortNumber(): string {
```

Comment:
nit: the function is called `newPortNumber()`, but returns a string?

Reply:
Yes, that's a quirk of me keeping the original function name (but earlier on switched to storing as a string, since that's how it gets used). I can rename.